api/repositories/pod_repository_test.go

```diff
@@ -0,0 +1,112 @@
+package repositories_test
+
+import (
+	apierrors "code.cloudfoundry.org/korifi/api/errors"
+	"code.cloudfoundry.org/korifi/api/repositories"
+	korifiv1alpha1 "code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+	"code.cloudfoundry.org/korifi/tests/matchers"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("PodRepository", func() {
+	var (
+		podRepo  *repositories.PodRepo
+		org      *korifiv1alpha1.CFOrg
+		space    *korifiv1alpha1.CFSpace
+		pod      *corev1.Pod
+		appGUID  string
+		app      *repositories.AppRecord
+		err      error
+		instance string
+		process  string
+	)
+	BeforeEach(func() {
+		instance = "2"
+		process = "web"
+		podRepo = repositories.NewPodRepo(userClientFactory)
+		org = createOrgWithCleanup(ctx, prefixedGUID("org"))
+		space = createSpaceWithCleanup(ctx, org.Name, prefixedGUID("space"))
+
+		appGUID = uuid.NewString()
+		app = &repositories.AppRecord{
+			GUID:      appGUID,
+			Revision:  "1",
+			SpaceGUID: space.Name,
+		}
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "podname-2",
+				Namespace: space.Name,
+				Labels: map[string]string{
+					"korifi.cloudfoundry.org/app-guid":     appGUID,
+					"korifi.cloudfoundry.org/version":      "1",
+					"korifi.cloudfoundry.org/process-type": "web",
+				}},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "web",
+						Image: "nginx",
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, pod)).To(Succeed())
+
+	})
+
+	JustBeforeEach(func() {
+		err = podRepo.DeletePod(ctx, authInfo, app.Revision, app.GUID, app.SpaceGUID, instance, process)
+	})
+
+	Describe("DeletePod", func() {
+		It("fails to delete the pod", func() {
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(matchers.WrapErrorAssignableToTypeOf(apierrors.ForbiddenError{}))
+		})
+		When("the user is a SpaceDeveloper", func() {
+			BeforeEach(func() {
+				createRoleBinding(ctx, userName, spaceDeveloperRole.Name, space.Name)
+			})
+
+			It("deletes the pod", func() {
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
We have to check that the pod object goes away eventually. Example: https://github.com/cloudfoundry/korifi/blob/e5f58770befffd0520e5f2fe52f574ce55ba4f69/api/repositories/domain_repository_test.go#L373-L376

Reply:
thanks for the example!